src/Microsoft.AspNetCore.JsonPatch/Internal/ListAdapter.cs

```diff
@@ -261,6 +266,14 @@ private bool TryGetPositionInfo(IList list, string segment, out PositionInfo pos
                     errorMessage = null;
                     return true;
                 }
+                // As per JSON Patch spec, for Add operation the index value representing the number of elements is valid,
+                // where as for other operations like Remove, Replace, Move and Copy the target index MUST exist.
+                else if (position == list.Count && operationType == OperationType.Add)
+                {
+                    positionInfo = new PositionInfo(PositionType.EndOfList, -1);
```

Comment:
Will this muck anything else up? Since we've now *lost* the fact that the index was provided, and the index was the end of the list, not `-1`